### DISCUSSION TITLE
Get All Workspace Symbols if search string is empty

### DISCUSSION THREAD

**Discussion by GollaSaiVenkatesh** [microsoft/pylance-release#5556] [Enhancement]
The  vscode.executeWorkspaceSymbolProvider is not returning all workspace symbols if query (search string) is empty when pylance is used as python language server.  This is useful for getting all symbols in the workspace.

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by MatejKastak** [comment]
This is also an issue with pyright. I am writing this as a follow up to https://github.com/microsoft/pyright/discussions/9305.

Based on the spec ([WorkspaceSymbolParams](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceSymbolParams)) for query parameter:
> A query string to filter symbols by. **Clients may send an empty string here to request all symbols.**

But the [implementation](https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/languageService/workspaceSymbolProvider.ts#L131-L135) is directly ignoring the spec.
> // Don't do a search if the query is empty. We'll return too many results in this case.

### Question
What should I do in this case if I want the full list, and I acknowledge that it might be too big?

Would it be possible to return the full list for the empty query, for users that opt-in (for example via config)? So that you can at least opt-in to spec compliant behavior?

### My motivation:
When searching for a symbol in the workspace, I would like to use filtering (fuzzy matching) from my editor, instead of the non-fuzzy matching of the query (which causes inconvenience when searching for symbols).
